### PR TITLE
feat: add p-mapbox-gl types

### PR DESCRIPTION
**Pull request by shaojun427**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by dmytro-gokun** [comment]
@shaojun427 What library this is for? Why did you add me here? :)

**Review by peterblazejewicz — CHANGES_REQUESTED** [review]
You need those people listed to give eplicitel consent to fact of being added in the header,
what's that library for? (it's appears to be just a fork)
thx!

**Comment by shaojun427** [comment]
> You need those people listed to give eplicitel consent to fact of being added in the header,
> what's that library for? (it's appears to be just a fork)
> thx!

This is a library based on mapbox-gl extension. I want to add a custom CRS parameter, but I haven’t added it yet. I want to replace the original mapbox with the fork library.

**Comment by shaojun427** [comment]
> @ shaojun427这是什么库？为什么在这里加我？:)

I'm sorry, this is a mapbox fork library. I forgot to modify some basic information when submitting it. Now your information has been deleted

**Comment by shaojun427** [comment]
> @shaojun427 What library this is for? Why did you add me here? :)

I'm sorry, this is a mapbox fork library. I forgot to modify some basic information when submitting it. Now your information has been deleted

File: types/p-mapbox-gl/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,2488 @@
+// Type definitions for Mapbox GL JS 1.0
+// Project: https://github.com/zsjjs/mapbox-gl-js
```
Comment (peterblazejewicz) [inline types/p-mapbox-gl/index.d.ts:2]:
```suggestion
// Project: https://github.com/zsjjs/p-mapbox-gl-js
```

File: types/p-mapbox-gl/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,2488 @@
+// Type definitions for Mapbox GL JS 1.0
```
Comment (peterblazejewicz) [inline types/p-mapbox-gl/index.d.ts:1]:
```suggestion
// Type definitions for p-mapbox-gl 1.0
```

**Comment by peterblazejewicz** [comment]
@shaojun427 if this is a fork and there is no real change in the API,
you can re-export eerything from Mapbox types IMO, just adding your changes.
That way it will be obvious what this package is about (extension based on a fork) and also maintenance burden could be lowered a lot.

**Review by peterblazejewicz — APPROVED** [review]
LGMT!

**Comment by elibarzilay** [comment]
The status of this PR is not clear to me: why is the listed owner different?  Why so many linter exceptions?

**Pull request closed without merging**